test/Validate.Storage.cpp

```diff
@@ -141,4 +141,34 @@ INSTANTIATE_TEST_CASE_P(MatrixOp, ValidateStorage,
                              "AtomicCounter",
                              "Image"));
 // clang-format on
+
+TEST_F(ValidateStorage, GenericVariableOutsideFunction) {
+  const auto str = R"(
+          OpCapability Kernel
+          OpMemoryModel Logical OpenCL
+%intt   = OpTypeInt 32 1
+%ptrt   = OpTypePointer Function %intt
+%var    = OpVariable %ptrt Generic
+)";
+  CompileSuccessfully(str);
+  ASSERT_EQ(SPV_ERROR_INVALID_LAYOUT, ValidateInstructions());
```

Comment:
Or the catchall INVALID_BINARY could work too.  Unless we have something more specific that is still accurate.